src/main/java/io/stargate/sgv2/jsonapi/api/model/command/Filterable.java

```diff
@@ -1,10 +1,18 @@
 package io.stargate.sgv2.jsonapi.api.model.command;
 
 import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.FilterClause;
+import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.FilterSpec;
 
 /*
  * All the commands that needs FilterClause will have to implement this.
  */
 public interface Filterable {
-  FilterClause filterClause();
+  /** Accessor for the filter specification in its intermediate for */
```

Comment:
I know I over engineer things :) but I wonder if there is a base interface here because this code looks like it will repeat

Reply:
Yeah, there's definitely a pattern of some kind. If we had "Deferred value" type (discussed above), we could have uber-interface here I think.
But not sure if that would help make code more readable.

Reply:
Added that new `JsonDefinition` type, but I think it's difficult to generalize this tag interface, since many commands have both `Filterable` and `Sortable`, so have to avoid name clash (for `JsonDefinition.json()`).
So I think this needs to remain separate from `Sortable`.